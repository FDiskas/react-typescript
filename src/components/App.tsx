import * as React from "react";

import { Demo } from "./Demo";

export interface IAppProps {
    title: string;
}

export class App extends React.Component<IAppProps, {}> {
    public render() {
        return <h1>Hello... <Demo title={this.props.title}/>from {this.props.title}!</h1>;
    }
}
