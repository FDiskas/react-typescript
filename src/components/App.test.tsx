import { shallow } from 'enzyme';

import enzymeToJson from 'enzyme-to-json';
import * as React from 'react';

import { App } from './App';

describe('App.js', () => {
    it('renders without crashing', () => {
        shallow(<App title="Test" />);
    });

    it('matches snapshot', () => {
        const wrapper = shallow(<App title="Test" />);
        expect(enzymeToJson(wrapper)).toMatchSnapshot();
    });

    it('should work', () => {
        const wrapper = shallow(<App title="Test" />);
        expect(enzymeToJson(wrapper)).toBeTruthy();
    });
});
