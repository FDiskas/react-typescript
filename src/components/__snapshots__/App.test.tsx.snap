// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.js matches snapshot 1`] = `
<h1>
  Hello... 
  <Demo
    title="Test"
  />
  from 
  Test
  !
</h1>
`;
